<a href="https://colab.research.google.com/github/pkdev0101/Flyrank-ML-internship-Pranay-Kamath/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/pkdev0101/Flyrank-ML-internship-Pranay-Kamath/blob/main/work/notebooks/w01_research_question.ipynb)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

I chose Lane 2: Refresh / Content Opportunity Scoring. I want to study which pseudonymized content pages should be reviewed first for a possible refresh, expansion, protection, pruning, or monitoring. I chose this lane because the starter notebooks showed that a learned ranking can find useful patterns beyond one simple hand-written rule. My unit of analysis will be one content page. My planned output will be a ranked review queue with a priority score, reason code, and suggested action.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

My research question is: Which content pages should a content editor review first based on observable search, freshness, and engagement signals?

This work improves the decision of which pages should enter a limited review queue first. A content editor or SEO analyst would inspect the recommended pages and decide whether to refresh, expand, protect, prune, or monitor them.

A wrong recommendation could waste an editor's time or cause an unnecessary change. Missing a useful candidate could allow a declining or underperforming page to go unnoticed. Data or ML may help because several signals c
an interact in ways that are difficult to combine with one fixed rule. The goal is to support a better decision, not just train a model.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [4]:
import os
import sys
import subprocess

REPO_URL = "https://github.com/pkdev0101/Flyrank-ML-internship-Pranay-Kamath"
REPO_DIR = "/content/Flyrank-ML-internship-Pranay-Kamath"

if "google.colab" in sys.modules:
    if not os.path.isdir(REPO_DIR):
        subprocess.run(
            ["git", "clone", "--depth", "1", REPO_URL, REPO_DIR],
            check=True
        )
    os.chdir(REPO_DIR)

import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

declining = df["trend_direction"].str.lower().eq("down")

stale_visible = (
    (df["days_since_last_update"] >= 180)
    & (df["impressions_90d"] >= 500)
)

low_ctr_visible = (
    (df["impressions_90d"] >= 500)
    & (df["avg_position"].between(1, 20))
    & (df["ctr"] < 0.5)
)

print(f"Total pages: {len(df):,}")
print(f"Declining pages: {declining.sum():,} ({declining.mean():.1%})")
print(f"Stale and visible pages: {stale_visible.sum():,}")
print(f"Visible pages with low CTR: {low_ctr_visible.sum():,}")

Total pages: 30,000
Declining pages: 16,262 (54.2%)
Stale and visible pages: 17
Visible pages with low CTR: 9,745


The starter dataset has 30,000 pages, and 16,262 of them, or 54.2%, are labeled as declining. Only 17 pages are both stale and highly visible under my current rule, suggesting that this rule may be too strict by itself. There are 9,745 visible pages with low CTR, showing that there are many possible review candidates. These numbers support using a ranked scoring system so an editor can focus on the strongest opportunities instead of reviewing thousands of pages equally.


---



## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This project can report measured associations and produce a directional, decision-support ranking. It can identify pages that appear to be stronger review candidates based on observed signals. It cannot prove that refreshing a page will cause traffic to recover, reveal Google's ranking algorithm, or guarantee that every highly ranked page should be changed. A human reviewer must inspect each recommendation before taking action.

---



## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.